src/stamp/modeling/data.py

```diff
@@ -189,30 +190,61 @@ def __getitem__(
             )
 
 
-def _get_coords_um(feature_h5: h5py.File) -> Tensor:
-    """Get coordinates in um from h5 file"""
-    coords = torch.from_numpy(feature_h5["coords"][:]).float()  # pyright: ignore[reportIndexIssue]
-    stride = cast(float, feature_h5.attrs.get("tile_size", get_stride(coords)))
+@dataclass
+class CoordsInfo:
+    coords_um: Tensor
+    tile_size_um: Microns
+    tile_size_px: SlidePixels | None
+
+    @property
+    def mpp(self) -> SlideMPP:
+        if not self.tile_size_px:
+            raise RuntimeError(
+                "tile size in pixels is not available. Please reextract them using `stamp preprocess`."
+            )
+        return SlideMPP(self.tile_size_um / self.tile_size_px)
 
-    if feature_h5.attrs.get("unit") == "um":
+
+def get_coords(feature_h5: h5py.File) -> CoordsInfo:
+    coords = torch.from_numpy(feature_h5["coords"][:]).float()  # pyright: ignore[reportIndexIssue]
+    coords_um: Tensor | None = None
+    tile_size_um: Microns | None = None
+    tile_size_px: SlidePixels | None = None
+    if (tile_size := feature_h5.attrs.get("tile_size", None)) and feature_h5.attrs.get(
+        "unit", None
+    ) == "um":
+        # STAMP v2 format
+        tile_size_um = Microns(tile_size)
         coords_um = coords
-    elif round(stride) == 224:
+    elif tile_size := feature_h5.attrs.get("tile_size_um", None):
+        # Newer STAMP format
+        tile_size_um = Microns(tile_size)
+        coords_um = coords
+    elif round(feature_h5.attrs.get("tile_size", get_stride(coords))) == 224:
+        # Historic STAMP format
         _logger.info(
             f"{feature_h5.filename}: tile stride is roughly 224, assuming coordinates have unit 256um/224px (historic STAMP format)"
         )
+        tile_size_um = Microns(256.0)
+        tile_size_px = SlidePixels(224)
```

Comment:
Please use `TilePixels`, like you suggested:)
```suggestion
        tile_size_px = TilePixels(224)
```